File: 백준/Gold/2015. 수들의 합 4/수들의 합 4.py
import sys
from collections import defaultdict
input = sys.stdin.readline
n,k = map(int, input().split())
A = list(map(int, input().split()))

pref = 0
freq = defaultdict(int)
freq[0] = 1
ans = 0

for x in A:
    pref += x
    ans += freq[pref-k]
    freq[pref] += 1

print(ans)

File: 백준/Gold/2015. 수들의 합 4/수들의 합 4.swift
import Foundation

let nk = readLine()!.split(separator: " ").map { Int($0)! }
let (n, k) = (nk[0], nk[1])

let arr = readLine()!.split(separator: " ").map { Int($0)! }
var pref = 0 
var ans = 0
var freq = [0:1]

for x in arr {
    pref += x
    ans += freq[pref-k, default:0]
    freq[pref, default:0] += 1
}
print(ans)